lib/active_model/serializer.rb

```diff
@@ -59,17 +59,11 @@ class << self
     end
 
     # @api private
-    def self.serializer_lookup_chain_for(klass)
-      chain = []
-
-      resource_class_name = klass.name.demodulize
-      resource_namespace = klass.name.deconstantize
-      serializer_class_name = "#{resource_class_name}Serializer"
-
-      chain.push("#{name}::#{serializer_class_name}") if self != ActiveModel::Serializer
-      chain.push("#{resource_namespace}::#{serializer_class_name}")
-
-      chain
+    def self.serializer_lookup_chain_for(klass, controller_class = nil)
+      lookups = ActiveModelSerializers.config.serializer_lookup_chain
+      Array[*lookups].flat_map do |lookup|
+        lookup.call(klass, self, controller_class)
```

Comment:
I'd rather just move all these class methods to a `ImplicitSerializer` utility module with a `lookup_by` method and a user-customizable strategy somewhere. e.g.

```ruby
config.serializer_lookup = ImplicitSerializer
```

Then someone can just change that to whatever they want and customize.

Now, it's still missing most of the controller context, .... or is it?  That's where `SerializationContext` comes it. It could take in a `namespace` option.